### PR TITLE
LPS 158155

### PR DESCRIPTION
**Pull request by robertoDiaz**

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

File: modules/apps/friendly-url/friendly-url-taglib/src/main/java/com/liferay/friendly/url/taglib/servlet/taglib/InputTag.java [hunk 1]
```diff
@@ -182,6 +182,21 @@ private String _getFallbackValue() {
...
+					if (isLocalizable()) {
+						return layout.getFriendlyURLsXML();
+					}
+
+					return layout.getFriendlyURL();
+				}
+
+				return StringPool.BLANK;
+			}
+
```
Comment (adolfopa) [inline modules/apps/friendly-url/friendly-url-taglib/src/main/java/com/liferay/friendly/url/taglib/servlet/taglib/InputTag.java:199]: Can you use `_getActualClassName` in the test for consistency?

Also, can you make conditional code a bit flatter? Nested conditionals are harder to understand. Something like:
```
if (Object.equals(_getActualClassname(), ...)) {
    Layout layout = ...;

    if (layout == null) {
        return StringPool.BLANK;
    }

    if (isLocalizable()) {
          ....
    }

    return layout.getFriendlyURLsXML();
```

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by AliciaGarciaGarcia** [comment]
I tested the solution with the dump of the mySQL provided and worked OK

Once you have addressed @adolfopa comments I think we can send it

**Review by AliciaGarciaGarcia — APPROVED** [review]

File: modules/apps/friendly-url/friendly-url-taglib/src/main/java/com/liferay/friendly/url/taglib/servlet/taglib/InputTag.java (hunk 1 above)
Comment (dacousalr) [reply modules/apps/friendly-url/friendly-url-taglib/src/main/java/com/liferay/friendly/url/taglib/servlet/taglib/InputTag.java:199]: > can you make conditional code a bit flatter? Nested conditionals are harder to understand

Sure thing.

> Can you use _getActualClassName in the test for consistency?

I would not mind, but given that the _getActualClassName makes some additional processing in its implementation precisely with the value to compare, the resulting test that still makes this fix work would be something like:

`if (StringUtil.startsWith(_getActualClassName(), Layout.class.getName())) {`

Would you be ok with this way?

Comment (adolfopa) [reply modules/apps/friendly-url/friendly-url-taglib/src/main/java/com/liferay/friendly/url/taglib/servlet/taglib/InputTag.java:199]: Oh, you're right. Then, we should use `getClassName` in both checks; it is pointless to keep using `_getActualClassName` inside `_getFallbackValue()` if it does the wrong thing.

**Comment by dacousalr** [comment]
Closing this to continue in https://github.com/liferay-lima/liferay-portal/pull/3273

**Pull request closed without merging**